Framework/HistogramData/test/HistogramRebinTest.h

```diff
@@ -0,0 +1,324 @@
+    BinEdges edges{0, 0.5, 1.5, 2};
+
+    auto outCounts = rebinCounts(hist, edges);
+    auto outFreq = rebinFrequencies(histFreq, edges);
+
+    TS_ASSERT_EQUALS(outCounts.y()[0], hist.y()[0] / 2);
+    TS_ASSERT_EQUALS(outCounts.y()[1], (hist.y()[0] + hist.y()[1]) / 2);
+    TS_ASSERT_EQUALS(outCounts.y()[2], hist.y()[1] / 2);
+
+    TS_ASSERT_EQUALS(outFreq.y()[0], histFreq.y()[0]);
+    TS_ASSERT_EQUALS(outFreq.y()[1], (histFreq.y()[0] + histFreq.y()[1]) / 2);
+    TS_ASSERT_EQUALS(outFreq.y()[2], histFreq.y()[1]);
+  }
+
+  void testCombineBinsAssymetric() {
+    // Handles the case where
+    // ||   ||   becomes:
+    // |  |  |
+    Histogram hist(BinEdges{0, 0.5, 1.5, 2}, Counts{10, 18, 7});
+    Histogram histFreq(BinEdges{0, 0.5, 1.5, 2}, Frequencies{16, 32, 8});
+    BinEdges edges{0, 1, 2};
+
+    auto outCounts = rebinCounts(hist, edges);
+    auto outFreq = rebinFrequencies(histFreq, edges);
+
+    TS_ASSERT_EQUALS(outCounts.y()[0], hist.y()[0] + (hist.y()[1] / 2));
+    TS_ASSERT_EQUALS(outCounts.y()[1], (hist.y()[1] / 2) + hist.y()[2]);
+
+    TS_ASSERT_EQUALS(outFreq.y()[0], (histFreq.y()[0] + histFreq.y()[1]) / 2);
+    TS_ASSERT_EQUALS(outFreq.y()[1], (histFreq.y()[1] + histFreq.y()[2]) / 2);
+  }
+
+  void testSplitCombineBinsAssymetric() {
+    // Handles the case where
+    // | | | |   becomes:
+    // ||   ||
+    Histogram hist(BinEdges{0, 1, 2, 3}, Counts{100, 50, 216});
+    Histogram histFreq(BinEdges{0, 1, 2, 3}, Frequencies{210, 19, 80});
+    BinEdges edges{0, 0.5, 2.5, 3};
+
+    auto outCounts = rebinCounts(hist, edges);
+    auto outFreq = rebinFrequencies(histFreq, edges);
+
+    TS_ASSERT_EQUALS(outCounts.y()[0], hist.y()[0] / 2);
+    TS_ASSERT_EQUALS(outCounts.y()[1],
+                     ((hist.y()[0] + hist.y()[2]) / 2) + hist.y()[1]);
+    TS_ASSERT_EQUALS(outCounts.y()[2], hist.y()[2] / 2);
+
+    TS_ASSERT_EQUALS(outFreq.y()[0], histFreq.y()[0]);
+    TS_ASSERT_EQUALS(
+        outFreq.y()[1],
+        ((histFreq.y()[0] / 2) + histFreq.y()[1] + (histFreq.y()[2] / 2)) / 2);
+    TS_ASSERT_EQUALS(outFreq.y()[2], histFreq.y()[2]);
+  }
+
+  void testSplitCombineBinsAssymetric2() {
+    // Handles the case where
+    // ||   ||   becomes:
+    // | | | |
+    Histogram hist(BinEdges{0, 0.5, 2.5, 3}, Counts{10, 100, 30});
+    Histogram histFreq(BinEdges{0, 0.5, 2.5, 3}, Frequencies{17, 8, 15});
+    BinEdges edges{0, 1, 2, 3};
+
+    auto outCounts = rebinCounts(hist, edges);
+    auto outFreq = rebinFrequencies(histFreq, edges);
+
+    TS_ASSERT_EQUALS(outCounts.y()[0], hist.y()[0] + (hist.y()[1] / 4));
+    TS_ASSERT_EQUALS(outCounts.y()[1], hist.y()[1] / 2);
+    TS_ASSERT_EQUALS(outCounts.y()[2], (hist.y()[1] / 4) + hist.y()[2]);
+
+    TS_ASSERT_EQUALS(outFreq.y()[0], (histFreq.y()[0] + histFreq.y()[1]) / 2);
+    TS_ASSERT_EQUALS(outFreq.y()[1], histFreq.y()[1]);
+    TS_ASSERT_EQUALS(outFreq.y()[2], (histFreq.y()[1] + histFreq.y()[2]) / 2);
+  }
```

Comment:
Maybe we should also add tests for start and end (the tests above cover the internal range):

```
|   |   |   |
  |   |   |
```

```
  |   |   |
|   |   |   |
```


```
|   |   |   |   |   |
        |   |
```
etc.?